x-pack/plugins/observability_solution/dataset_quality/public/state_machines/dataset_quality_controller/src/state_machine.ts

```diff
@@ -132,18 +134,55 @@ export const createPureDatasetQualityControllerStateMachine = (
         flyout: {
           initial: 'closed',
           states: {
-            fetching: {
-              invoke: {
-                src: 'loadDataStreamDetails',
-                onDone: {
-                  target: 'loaded',
-                  actions: ['storeDatasetDetails'],
+            initializing: {
+              type: 'parallel',
+              states: {
+                dataStreamDetails: {
+                  initial: 'fetching',
+                  states: {
+                    fetching: {
+                      invoke: {
+                        src: 'loadDataStreamDetails',
+                        onDone: {
+                          target: 'done',
+                          actions: ['storeDatasetDetails'],
+                        },
+                        onError: {
+                          target: 'done',
+                          actions: ['fetchDatasetDetailsFailedNotifier'],
+                        },
+                      },
+                    },
+                    done: {
+                      type: 'final',
+                    },
+                  },
                 },
-                onError: {
-                  target: 'loaded',
-                  actions: ['fetchDatasetDetailsFailedNotifier'],
+                integrationDashboards: {
```

Comment:
I see your point, just a bit worried about the fact that this information is being queried every time that a user open a flyout when it might not be needed